components/script/dom/gpudevice.rs

```diff
@@ -523,9 +529,74 @@ impl GPUDeviceMethods for GPUDevice {
                 descriptor.layout.id(),
                 bindings,
             ))
-            .expect("Failed to create WebGPU PipelineLayout");
+            .expect("Failed to create WebGPU BindGroup");
 
         let bind_group = receiver.recv().unwrap();
         GPUBindGroup::new(&self.global(), bind_group, valid)
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createshadermodule
+    fn CreateShaderModule(
+        &self,
+        descriptor: RootedTraceableBox<GPUShaderModuleDescriptor>,
+    ) -> DomRoot<GPUShaderModule> {
+        let (sender, receiver) = ipc::channel().unwrap();
+        let program: Vec<u32> = match &descriptor.code {
+            Uint32Array(program) => program.to_vec(),
+            String(program) => program
+                .as_bytes()
+                .iter()
+                .map(|c| *c as u32)
+                .collect::<Vec<u32>>(),
+        };
+        let id = self
+            .global()
+            .wgpu_create_shader_module_id(self.device.0.backend());
+        self.channel
+            .0
+            .send(WebGPURequest::CreateShaderModule(
+                sender,
+                self.device,
+                id,
+                program,
+            ))
+            .expect("Failed to create WebGPU ShaderModule");
+
+        let shader_module = receiver.recv().unwrap();
```

Comment:
In all of the creation methods, we aren't expected to block on anything. Here, therefore, we can't `recv()`. Instead, we'd be creating a new ID and asking the GPU server side to create something associated with this ID, but never wait for it to do so.

Reply:
Yes, We have an issue ( #25472 ) for this, we would like to address this in a separate pull request for all the call which has to be non-blocking.